src/sentry/notifications/notifications/strategies/member_write_role_recipient_strategy.py

```diff
@@ -10,9 +10,20 @@
 
 class MemberWriteRoleRecipientStrategy(RoleBasedRecipientStrategy):
     def determine_member_recipients(self) -> Iterable[OrganizationMember]:
+        valid_roles = (r.id for r in roles.get_all() if r.has_scope("member:write"))
```

Comment:
@cathyteng17 Feels like we could push some of this logic to `RoleBasedRecipientStrategy` to avoid having to replicate this logic 3x in Sentry. Every subclass of this has pretty similar logic, just need to set some sort of field for which roles/scopes we need.

Reply:
sounds good to me